### PR TITLE
CVSL-2264: Add endpoint to update case admin additional information o…

### PR DESCRIPTION
**Pull request by justin-gardiner**
…n an address check request.

### CODE REVIEW

**Review by SebNegahbanMoJ — DISMISSED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/prison/AddressService.kt [hunk 1]
```diff
@@ -159,12 +142,36 @@ class AddressService(
       dateOfBirth = addResidentRequest.dateOfBirth,
       age = addResidentRequest.age,
       isMainResident = addResidentRequest.isMainResident,
-      standardAddressCheckRequest = standardAddressCheckRequest,
+      standardAddressCheckRequest = standardAddressCheckRequest as StandardAddressCheckRequest,
```
[the remark this answers is not in the document]
Comment (justin-gardiner) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/prison/AddressService.kt]: Good point, done.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/prison/AddressService.kt [hunk 2]
```diff
@@ -159,12 +142,44 @@ class AddressService(
...
+  ) {
+    val curfewAddressCheckRequest = getCurfewAddressCheckRequest(requestId, prisonNumber)
+    curfewAddressCheckRequest.caAdditionalInfo = caseAdminInfoRequest.additionalInformation
+    curfewAddressCheckRequestRepository.save(curfewAddressCheckRequest)
+  }
+
+  private fun getCurfewAddressCheckRequest(requestId: Long, prisonNumber: String): CurfewAddressCheckRequest {
+    val curfewAddressCheckRequest =
+      curfewAddressCheckRequestRepository.findByIdOrNull(requestId)
+        ?: throw EntityNotFoundException("Cannot find standard address check request with id: $requestId")
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/prison/AddressService.kt:165]: guess these messages aren't quite right as they could be any type of address check?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/prison/AddressService.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (justin-gardiner) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/prison/AddressService.kt]: Done

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**